Migrate foundation website content

brianwarner opened the pull request:
This is a work-in-progress, please do not merge until the `On hold` label is removed.

## Description

I am working on migrating and integrating the content from https://foundation.graphql.org into the main site. This will help with discoverability and maintenance, and will hopefully also raise the visibility of the Foundation to graphql.org visitors. Membership is the single source of funding for the Foundation, and is how the Foundation is able to provide community support.

There will be more commits coming, particularly with regards to the blog, newsletters, and annual report. I'm planning to do quite a few reviews before suggesting this be merged, as it does reorganize a few other things as well.

The patches should apply incrementally, and they also provide some clarifications on existing content. Where possible I've attempted to reuse existing templates and CSS, although there were a few changes that required new changes. In that case, I split the CSS directives into separate sections/files. None of the changes should be breaking (or if they are, it wasn't intentional).

Other fixes included:
* #1022 

Thank you!

brianwarner committed:
Add FAQ item on specification membership

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add FAQ section on the GraphQL Foundation

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Update the URL to the landscape

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Update users page to use new URL and borderless cards

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add Foundation members landscape cards

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add foundation logos

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add layout and css pre-reqs for foundation site

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add tweaks for foundation-specific pages

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add initial Foundation pages

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Move CoC to footer, add to community page

The header menu had too many items and was causing strange wrapping with the Algolia search box.
This moves the link to the CoC to the footer in a logical place, and also adds a link to the
Community Resources page.

It also adds a link to GitHub to the community page.

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Fix link to super-graph (now graphjin)

Signed-off-by: Brian Warner <brian@bdwarner.com>

saihaj commented:
> there's a way to define the raw iframe as a component, and then import it into a .md files in src/content?

If this was using [MDX](https://www.gatsbyjs.com/docs/how-to/routing/mdx/) then you can import a components in a Markdown File

You can also modify the `DocsSidebar` component so it takes optional props and then pass additional sidebar categories from `FoundationLayout`

brianwarner commented:
Thanks @saihaj!

I do like the first idea, of defining the iframe as a component and importing it. The html should really never change, but the text above it or the menu definition might, and it could be more maintainable if the pages are always maintained the same way. I am still pretty new to Gatsby and React though, so I'll have to do some learning to figure out how to do this.

Are there any negative consequences to adding the MDX plugin? Could this introduce any other unexpected consequences elsewhere on the site?

saihaj commented:
I don't think it should cause any issues. Here are gatsby docs on [migrating from remark to mdx](https://www.gatsbyjs.com/docs/how-to/routing/migrate-remark-to-mdx/)

orta committed:
Adds a members sidebar, and cleans up gatsby-node

orta committed:
Merge branch 'migrate-foundation-website-content' of https://github.com/graphql/graphql.github.io into migrate-foundation-website-content

orta commented:
Yeah, switching to MDX is a big thing - let's avoid that

I've shipped a change which adds  bunch of comments to the launch process, and adds the ability to arbitrarily add links to different sections of the site including the one you asked for 👍🏻 

![Screen Shot 2021-03-09 at 9 50 52 PM](https://user-images.githubusercontent.com/49038/110543037-b4abe480-8121-11eb-8d89-f74e8904cc94.png)

brianwarner commented:
@orta this is **great**! Thank you so much for having a look at it and for the help. I have some more things I'd like to add before it's ready for review, but this definitely gets me unstuck on that one part.

brianwarner committed:
Add Twitter card plugin

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add annual report, reorganize pages

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Update broken link

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add missing membership agreement preview copies

Signed-off-by: Brian Warner <brian@bdwarner.com>

orta committed:
Adds blog excerpts in index

orta commented:
The blog index now shows an excerpt:

![Screen Shot 2021-03-12 at 1 34 15 PM](https://user-images.githubusercontent.com/49038/110947270-daaad200-8337-11eb-8e71-49a53aee54a9.png)

brianwarner committed:
Add blog posts from Foundation site

This commit adds a bunch of content from the Foundation site. It also adjusts the CSS to make each
post a little easier to see in the sidebar, and adds tags to each post for (hopefully) future
support of blog categories.

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add link to blog; make formatting consistent

Signed-off-by: Brian Warner <brian@bdwarner.com>

benjie commented:
@brianwarner Might be worth adding `Fixes #1022` to the issue description :+1:

brianwarner committed:
Restructure and expand the community section

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Reorganize and expand the footer

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Merge branch 'source' into migrate-foundation-website-content

brianwarner commented:
Ok, I think I'm ready for a review!

I added quite a bit of content in the community section. We really didn't have much on how the project itself actually operates, or where to find working groups, or what stuff the community can expect from the Foundation.

I'd appreciate a review on the following:
* The Foundation sub-site in general: https://deploy-preview-1019--graphql-org.netlify.app/foundation/
* Foundation FAQs: https://deploy-preview-1019--graphql-org.netlify.app/faq/graphql-foundation
* Fixes to the layout of users: https://deploy-preview-1019--graphql-org.netlify.app/users
* A bunch of new content in community: https://deploy-preview-1019--graphql-org.netlify.app/community
* Changes to the footer links: https://deploy-preview-1019--graphql-org.netlify.app/

I also temporarily fixed the search bar overflow problem referenced in a few issues, though I see there's an open PR that adds a hamburger menu. It would be nice to merge that afterwards.

Looking for reviews in general, but particularly from @leebyron @IvanGoncharov @orta @carolstran @caniszczyk @Urigo @saihaj @benjie

saihaj commented on gatsby-config.js:
```diff
@@ -16,7 +16,19 @@ module.exports = {
+      options: {
+        plugins: [
+          {
+            resolve: "@weknow/gatsby-remark-twitter",
+            options: {
+              debug: true
```
where are the tweet card rendered by this plugin in foundation blog posts? I tried to look around but couldn't find one the embedded tweet only links that take to tweets.

brianwarner replied:
This is used to render the tweet cards on the 2019 annual report: https://deploy-preview-1019--graphql-org.netlify.app/foundation/annual-reports/2019/#attendee-feedback

Basically if you go to a tweet and copy Twitter's embed code, this renders the card even on a pure markdown page. Without it, it merely presents the text of the tweet (but in a nicely formatted blockcode, so it's still a great fallback).

saihaj commented on src/components/BlogPost/index.tsx:
```diff
@@ -1,3 +1,4 @@
+import { printIntrospectionSchema } from "graphql"
```
unused import

brianwarner replied:
Good catch, removing this now.

saihaj commented on src/components/FoundationLayout/index.tsx:
```diff
@@ -0,0 +1,48 @@
+  pageContext: any
+}
+
+const index = ({ title, heroText, nextDoc, sideBarData, rawMarkdownBody, pageContext }: Props) => {
```
Just to be consistent with component names in other files
```suggestion
const Index = ({ title, heroText, nextDoc, sideBarData, rawMarkdownBody, pageContext }: Props) => {
```

brianwarner replied:
Thank you, this is in the commit I just pushed.

saihaj commented on src/components/FoundationLayout/index.tsx:
```diff
@@ -0,0 +1,48 @@
+  )
+}
+
+export default index
```
```suggestion
export default Index
```

brianwarner replied:
And this is in the most recent commit as well.

saihaj reviewed (changes requested):
Great work 👍🏼 finally the `/blog` is more discoverable and restructure of `/community` with many new resources.

This page looks broken https://deploy-preview-1019--graphql-org.netlify.app/faq/graphql-foundation it is rendering perfectly how it is suppose in `/faq` I guess this is something which can be handled in new issue so pages are not generated for FAQ sections since every Faq has a page generated which on visiting just feels broken.


Image is not being rendered here https://deploy-preview-1019--graphql-org.netlify.app/blog/2020-06-30-gsoc-2020-participant-naman exploring this further found that there is SVG being used so we possible need a way to handle SVG's in markdown as images `gatsby-remark-images` doesn't do it https://github.com/gatsbyjs/gatsby/issues/7317#issuecomment-412984851

![Google Summer of Code 2020](https://user-images.githubusercontent.com/44710980/111240924-22488c80-85ca-11eb-8ed7-6e5c7312c339.png)

brianwarner commented:
Good point on the FAQ, here's a better link: https://deploy-preview-1019--graphql-org.netlify.app/faq/#what-does-the-graphql-foundation-do

brianwarner committed:
Fix broken link, file was not migrated

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner commented:
Also good catch on the .svg, I just didn't move the file over. That's fixed now.

brianwarner committed:
Remove unused import

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Fix capitalization for consistency

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner commented:
I should also have noted - in Slack I mentioned that I haven't been able to get blog tags working. All of the posts are tagged, so once we get there it will be supported, it just took a little more knowledge of the site than I have at the moment.

orta commented on src/components/Footer/index.tsx:
```diff
@@ -47,26 +55,18 @@ const getLinks = (sourcePath: string): FooterLinks[] => [
+      { text: "Artwork", href: "https://github.com/graphql/artwork" },
+      { text: "Contact", href: "/foundation/contact/" },
     ],
   },
```
Nice work on these changes

orta commented on src/pages/users.tsx:
```diff
@@ -30,7 +30,7 @@ export default ({ pageContext }) => {
           scrolling="no"
           style={{ width: "1px", minWidth: "100%" }}
-          src="https://landscape.graphql.org/category=graph-ql-adopter&format=card-mode&grouping=category&embed=yes"
+          src="https://landscape.graphql.org/card-mode?category=graph-ql-adopter&grouping=category&embed=yes&style=borderless"
```
fixes https://github.com/graphql/graphql.github.io/pull/1011

orta reviewed (approved):
Ran throughout locally with the latest source` merged in and it all feels good to me 👍🏻

benjie commented:
Not sure what's going on here, but loading https://deploy-preview-1019--graphql-org.netlify.app/faq/graphql-foundation in both Firefox and Chrome, none of the FAQ sections are expandable. There's no error in the JS console :shrug:

benjie commented:
On the Users page:

![Screenshot_20210316_085532](https://user-images.githubusercontent.com/129910/111281978-6624ab80-8635-11eb-9d15-1d47c36aec04.png)

benjie commented on src/content/community/Community-Developers.md:
```diff
@@ -0,0 +1,86 @@
+* The official [GraphQL Slack](https://slack.graphql.org)
+* Instructions on how to sign the [GraphQL Specification Membership Agreement](https://github.com/graphql/graphql-wg/tree/HEAD/membership)
+* The [GraphQL project calendar](https://calendar.graphql.org)
+* Meeting replays on the [GraphQL YouTube channel](https://youtube.graphql.org)
```
Handy redirect! :+1:

benjie commented on src/content/community/Community-Developers.md:
```diff
@@ -0,0 +1,86 @@
+
+#### The GraphQL Working Group
+
+The [GraphQL Working Group](https://github.com/graphql/graphql-wg) is the main group that plans and implements changes to the [GraphQL specification](https://spec.graphql.org). It meets actively and is the default place for discussion for topics which do not have a separate breakout working group (e.g., input unions, graphql-over-http, etc).
```
The `e.g.` here is ambigous; it could imply that these topics don't have separate breakout working groups. I've clarified.

```suggestion
The [GraphQL Working Group](https://github.com/graphql/graphql-wg) is the main group that plans and implements changes to the [GraphQL specification](https://spec.graphql.org). It meets actively and is the default place for discussion for topics which do not have a separate breakout working group (breakout working groups include the [Input Union WG](https://github.com/graphql/graphql-spec/blob/main/rfcs/InputUnion.md) and the [GraphQL-over-HTTP WG](https://github.com/graphql/graphql-over-http)).
```

benjie commented on src/content/community/Community-Developers.md:
```diff
@@ -0,0 +1,86 @@
+
+### Financial and policy governance
+
+The [GraphQL Foundation](/foundation/) provides financial support for the GraphQL project. In keeping wiht best practices, the Foundation is a separate organization, which helps to ensure that technical decisions are made upon their own merit and independent of financial contributions. The GraphQL Foundation uses the funds raised through [membership dues](/foundation/join/) to support the work of ...
```
```suggestion
The [GraphQL Foundation](/foundation/) provides financial support for the GraphQL project. In keeping with best practices, the Foundation is a separate organization, which helps to ensure that technical decisions are made upon their own merit and independent of financial contributions. The GraphQL Foundation uses the funds raised through [membership dues](/foundation/join/) to support the work of the GraphQL Project through investments in infrastructure, coordination support services, developer grants, and mentorship programs.
```

benjie commented on src/content/community/Community-Resources.md:
It's odd that these link to different places. I feel like it's deliberate, but it's also unexpected, so maybe we should split them up:

```suggestion
* **Slack:** [slack.graphql.org](https://slack.graphql.org) ([get an invite](https://slack-invite.graphql.org))
```

brianwarner replied:
Works for me!

benjie commented on src/content/community/Community-RunningProjects.md:
```diff
@@ -0,0 +1,79 @@
+> 
+> If you maintain a tool or reference implementation that you'd like to transfer to the [GraphQL GitHub org](https://github.com/graphql), or if you are planning to start a new workgroup or implementation within the official GraphQL project, please add your name to the agenda of an upcoming [GraphQL Working Group meeting](https://github.com/graphql/graphql-wg/tree/HEAD/agendas) to introduce your i ...
+> 
+> A few things will then need to happen. For example, we will work with you to adopt our policies and neutral governance, and help you transfer the project and its assets (generally the domain name and any trademarks) so that they are neutrally held at the Linux Foundation. If you'd like to explore the idea before making a proposal, please reach out to [operations@graphql.org](mailto:operations@g ...
```
This makes it feel like you're definitely going to be accepted; I've caveated it:

```suggestion
> Should your proposal meet with approval, a few things will then need to happen. For example, we will work with you to adopt our policies and neutral governance, and help you transfer the project and its assets (generally the domain name and any trademarks) so that they are neutrally held at the Linux Foundation. If you'd like to explore the idea before making a proposal, please reach out to [operations@graphql.org](mailto:operations@graphql.org)
```

benjie replied:
Further, is it correct that the Linux Foundation holds the assets, rather than the GraphQL Foundation? I'm guessing it is correct due to the way the GF/LF relate, but wanted to make sure it wasn't a copy/paste error just in case.

brianwarner replied:
Yep, that's accurate. LF is the actual legal entity and thus the one that holds things like trademarks, whereas GraphQL Foundation is technically a fund within the LF.

benjie commented on src/content/community/Community-RunningProjects.md:
```diff
@@ -0,0 +1,79 @@
+
+### Trademark guidance
+
+The GraphQL trademark is managed by LF Projects. GraphQL projects are able to use the GraphQL name and marks in accordance with the [trademark policy](https://lfprojects.org/policies/trademark-policy/). If your project has questions about how you or others can use the mark, please contact [trademark@graphql.org](mailto:trademark@graphql.org)
```
```suggestion
The GraphQL trademark is managed by LF Projects. GraphQL projects are able to use the GraphQL name and marks in accordance with the [trademark policy](https://lfprojects.org/policies/trademark-policy/). If your project has questions about how you or others can use the mark, please contact [trademark@graphql.org](mailto:trademark@graphql.org).
```

benjie commented on src/content/faq/Foundation.md:
```diff
@@ -0,0 +1,55 @@
+The GraphQL Foundation is funded exclusively through the support of our members. The governing board sets an annual budget to use the dues to the maximum benefit of the community.
+Ways the Foundation distributes the money raised through memberships include:
+
+* Providing grants and sponsorship to core community developers who are working in neutral roles.
+* Funding core community IT infrastructure
+* Funding coordination, financial, and legal support program for the Foundation and the GraphQL project ecosystem
+* Other community and developer support programs
```
```suggestion
* Providing grants and sponsorship to core community developers who are working in neutral roles.
* Funding core community IT infrastructure.
* Funding coordination, financial, and legal support programs for the Foundation and the GraphQL project ecosystem.
* Other community and developer support programs.
```

benjie commented on src/content/foundation/Contact.md:
```diff
@@ -0,0 +1,49 @@
+
+## Membership Support
+
+If you are already a member of the [GraphQL Foundation](./) and you need help with your membership, please reach out to [operations@graphql.org](mailto:operations@graphql.org).
```
Cannot use relative link here since this page `https://deploy-preview-1019--graphql-org.netlify.app/foundation/contact` auto-redirects to `https://deploy-preview-1019--graphql-org.netlify.app/foundation/contact/`, so `./` links to itself.

```suggestion
If you are already a member of the [GraphQL Foundation](/foundation/) and you need help with your membership, please reach out to [operations@graphql.org](mailto:operations@graphql.org).
```

brianwarner replied:
Good catch, that was a copy and paste error.

benjie reviewed (changes requested):
Looking good; only minor tweaks needed

brianwarner commented:
> Not sure what's going on here, but loading https://deploy-preview-1019--graphql-org.netlify.app/faq/graphql-foundation in both Firefox and Chrome, none of the FAQ sections are expandable. There's no error in the JS console 🤷

This was my fault, I didn't realize that the FAQ items don't expand if you link straight to a section. Try the full FAQ page, instead: https://deploy-preview-1019--graphql-org.netlify.app/faq/

brianwarner commented:
> On the Users page:
> 
> ![Screenshot_20210316_085532](https://user-images.githubusercontent.com/129910/111281978-6624ab80-8635-11eb-9d15-1d47c36aec04.png)

This comes from the embedded landscape view (an iframe). I'm talking with the landscape developer and it sounds like they're working on the ability to pass custom CSS through to the landscape view. I assume in the process the `undefined` divs will get classes and/or IDs.

brianwarner committed:
Update src/content/community/Community-Developers.md

Improvements and clarifications from Benjie

Signed-off-by: Brian Warner <brian@bdwarner.com>

Co-authored-by: Benjie Gillam <benjie@jemjie.com>

brianwarner committed:
Update src/content/community/Community-Developers.md

Spelling fix from Benjie, thank you!

Signed-off-by: Brian Warner <brian@bdwarner.com>

Co-authored-by: Benjie Gillam <benjie@jemjie.com>

brianwarner committed:
Update src/content/community/Community-Resources.md

Improvement from Benjie, fixing links that were confusing.

Signed-off-by: Brian Warner <brian@bdwarner.com>

Co-authored-by: Benjie Gillam <benjie@jemjie.com>

brianwarner committed:
Update src/content/community/Community-RunningProjects.md

Clarification on process, thanks Benjie.

Signed-off-by: Brian Warner <brian@bdwarner.com>

Co-authored-by: Benjie Gillam <benjie@jemjie.com>

brianwarner committed:
Update src/content/community/Community-RunningProjects.md

Punctuation fix, thanks Benjie.

Signed-off-by: Brian Warner <brian@bdwarner.com>

Co-authored-by: Benjie Gillam <benjie@jemjie.com>

brianwarner committed:
Update src/content/faq/Foundation.md

Grammar fix, thanks Benjie.

Signed-off-by: Brian Warner <brian@bdwarner.com>

Co-authored-by: Benjie Gillam <benjie@jemjie.com>

brianwarner committed:
Update src/content/foundation/Contact.md

Fix copy and paste error from moving the Foundation site off of its own subdomain. Thanks Benjie.

Signed-off-by: Brian Warner <brian@bdwarner.com>

Co-authored-by: Benjie Gillam <benjie@jemjie.com>

benjie commented on src/content/community/Community-RunningProjects.md:
```diff
@@ -0,0 +1,79 @@
+
+The **person running the call** must:
+
+* Merge all PRs with passing checks prior to the meeting.
```
If this were a legal document... This would be saying that if you run a call you have to merge all PRs with passing checks, which could involve PRs such as "deleting all files". This doesn't feel quite right... but I may be overthinking it.

benjie replied:
Maybe something like:

```suggestion
* Merge all appropriate attendance and agenda PRs with passing checks prior to the meeting.
```

brianwarner replied:
Ha, yes, good point. I like this change.

benjie reviewed (approved):
LGTM :+1:

orta reviewed (approved):
Looks good to me

Urigo commented on src/content/community/Community-Events.md:
```diff
@@ -3,18 +3,55 @@ title: Upcoming Events
+All GraphQL project meetings should appear on the [GraphQL public calendar](https://calendar.graphql.org). If a project's meeting isn't on the calendar or is incorrect, please email [operations@graphql.org](mailto:operations@graphql.org).
 
-GraphQL conference organized by Hasura
+* **Who can join GraphQL project meetings:** Anyone who has signed the free [GraphQL Specification Membership agreement](https://github.com/graphql/graphql-wg/tree/HEAD/membership).
```
So great to highlight the meeting here!

Urigo commented on src/content/community/Community-Users.md:
Not related and has nothing to do with the review so if it's a problem we'll add it later but maybe you could add also The Guild's Discord channel here?

https://discord.gg/xud7bH9

Also worth mentioning that the link to Discord in the section above makes it look like Reactiflux is the official GraphQL Discord channel which is really not, it's just a small section on their Discord

benjie replied:
Yeah, I wondered what the criteria for getting listed here is! If you're adding community chats, it's https://discord.gg/graphile for the Graphile Discord :+1:

brianwarner replied:
Yes, let's add both! The main thing is that I'd like to ensure we're clear which is a general community channel, and which are run by organizations in the ecosystem. Just so there's no confusion.

Urigo replied:
I think in terms of Discord, there is no general community, so maybe put all Discord channels together under the Vendor section?
For me the Reactiflux one is also a "Vendor" but I might be wrong so maybe someone else can confirm

Urigo reviewed (approved):
This is so great!
Sorry it took me a while to look at this, I already see how much value it adds to the website!
Can't wait to have it live!

none of my comments are related to the changes and are not blocking and/or important.

P.S.
I see that the CI on Netlify has been broken on `source` a couple of commits ago, I guess you are aware of that?
Ohh I see now you actually solved it here :) `dosco/super-graph`, nice!

brianwarner committed:
Update src/content/community/Community-RunningProjects.md

Signed-off-by: Brian Warner <brian@bdwarner.com>

Co-authored-by: Benjie Gillam <benjie@jemjie.com>
